src/toil/batchSystems/mesos/executor.py

```diff
@@ -212,11 +233,75 @@ def frameworkMessage(self, driver, message):
         log.debug("Received message from framework: {}".format(message))
 
 
-def main(executorClass=MesosExecutor):
+def main():
     logging.basicConfig(level=logging.DEBUG)
     log.debug("Starting executor")
-    executor = executorClass()
-    driver = mesos.native.MesosExecutorDriver(executor)
-    exit_value = 0 if driver.run() == mesos_pb2.DRIVER_STOPPED else 1
+
+    if not os.environ.has_key("MESOS_AGENT_ENDPOINT"):
+        # Some Mesos setups in our tests somehow lack this variable. Provide a
+        # fake one to maybe convince the executor driver to work.
+        os.environ["MESOS_AGENT_ENDPOINT"] = os.environ.get("MESOS_SLAVE_ENDPOINT", "127.0.0.1:5051")
+        log.warning("Had to fake MESOS_AGENT_ENDPOINT as %s" % os.environ["MESOS_AGENT_ENDPOINT"])
+        
+    try:
+        urlopen("http://%s/logging/toggle?level=1&duration=15mins" % os.environ["MESOS_AGENT_ENDPOINT"]).read()
+        log.debug("Toggled agent log level")
+    except Exception as e:
+        log.debug("Failed to toggle agent log level")
+        
+    # Parse the agent state
+    agent_state = json.loads(urlopen("http://%s/state" % os.environ["MESOS_AGENT_ENDPOINT"]).read())
+    if agent_state.has_key('completed_frameworks'):
+        # Drop the completed frameworks whichg grow over time
+        del agent_state['completed_frameworks']
+    log.debug("Agent state: %s", str(agent_state))
+    
+    log.debug("Virtual memory info in executor: %s" % repr(psutil.virtual_memory()))
+    
+    if os.path.exists('/sys/fs/cgroup/memory'):
+        # Mesos can limit memory with a cgroup, so we should report on that.
+        for (dirpath, dirnames, filenames) in os.walk('/sys/fs/cgroup/memory', followlinks=True):
+            for filename in filenames:
+                if 'limit_in_bytes' not in filename:
+                    continue
+                log.debug('cgroup memory info from %s:' % os.path.join(dirpath, filename))
+                try:
+                    for line in open(os.path.join(dirpath, filename)):
+                        log.debug(line.rstrip())
+                except Exception as e:
+                    log.debug("Failed to read file")
+                    
+    # Mesos can also impose rlimit limits, including on things that really
```

Comment:
Sorry if it's a dumb question, but do you know why it does this, or the limiting by cgroup?

Reply:
Mesos applies rlimits to jobs to keep them from using more memory than they asked for and interfering with other scheduled jobs on the same node. It's one of the containerization/isolation strategies that you can have it use instead of cgroups, which I don't think you can do without root, at least on a lot of deployed kernels.